### DISCUSSION TITLE
how to install from China? (with a vpn? via ssh on a raspberry pi)

### DISCUSSION THREAD

**Discussion by philou155** [tphakala/birdnet-go#326] [Q&A]
Trying to install birdnet-go from China and it's not working... I could install docker on my raspberry pi 4 (on Raspberry Pi OS Lite 64-bit, Debian Bookworm) via ssh but the next step is not working and i think it's because it might be blocked in China

> ~ $ docker run -ti -p 8080:8080 --device /dev/snd ghcr.io/tphakala/birdnet-go:latest
> docker: unknown server OS: .
> See 'docker run --help'.

then i tried the "[New install script for simplified Docker image deployment](https://github.com/tphakala/birdnet-go/discussions/325)" and it seem to start the install but it then stopped:

> 🐳 Pulling BirdNET-Go Docker image from GitHub Container Registry...
> permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.47/images/create?fromImage=ghcr.io%2Ftphakala%2Fbirdnet-go&tag=dev": dial unix /var/run/docker.sock: connect: permission denied
> ❌ Failed to pull Docker image
> This could be due to:
> - No internet connection
> - GitHub container registry being unreachable
> - Invalid image name or tag

I'm on a windows machine using windows powershell and i tried to turn on my vpn (VPNGUI) but it doesn't seem to help...

is there a way to tunnel my ssh install with my vpn or should i try to use another install method?
thanks!

Philippe

**Comment by tphakala** [comment]
```
"http://%2Fvar%2Frun%2Fdocker.sock/v1.47/images/create?fromImage=ghcr.io%2Ftphakala%2Fbirdnet-go&tag=dev": dial unix /var/run/docker.sock: connect: permission denied
```

This indicates that your user account does not have privileges required to use docker, install.sh takes care of this, although I just realized that it checks group membership only when docker is installed with install.sh, not if docker is already available. I need to fix this.

Add user user as member of docker group
```
sudo usermod -aG docker $USER
```

And log out of Linux shell session and log in to make group membership change to take effect and run install.sh again, this time it should work.

**Reply by tphakala** [reply]
I updated install.sh to check Docker group membership, can you @philou155 please try again with updated install.sh?

**Comment by philou155** [comment]
ah ok thanks! I tried again and got this:

> $ curl -fsSL https://github.com/tphakala/birdnet-go/raw/main/install.sh -o install.sh
> curl: (28) Failed to connect to github.com port 443 after 133586 ms: Couldn't connect to server

**Reply by tphakala** [reply]
This is an issue with your intenet connection, you need access to github.com etc. in order to install BirdNET-Go. I can't help other than ask you to try with VPN etc.

**Comment by philou155** [comment]
ok thank you, I tried again with vpn on and this time it worked with docker (~ $ docker run -ti -p 8080:8080 --device /dev/snd ghcr.io/tphakala/birdnet-go:latest)

now, coming from birdnet-pi, I'm a bit lost... I tried to go to localhost:8080 but couldn't. Looked at the wiki guide, but seems a lot more technical.

> 2024/12/11 16:22:58 Using latest range filter model
> System details: linux debian 12.5 on  hardware
> Starting analyzer in realtime mode. Threshold: 0.8, overlap: 1.5, sensitivity: 1, interval: 15
> Logging disabled
> 2024/12/11 16:22:58 Clip retention policy: usage
> ⇨ http server started on [::]:8080
> Available Capture Sources:
>   0: Discard all samples (playback) or generate zero samples (capture), null
> 2024/12/11 16:22:58 Error selecting capture source: no suitable capture source found for device setting sysdefault

(I didn't plug the mic when installing so i think that's normal)

also I only have a raspberry pi 4B with 1GB of RAM. hopefully I can still test it a bit.
